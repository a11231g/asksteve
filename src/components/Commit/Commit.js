import React, { PureComponent } from 'react';
import {
    View,
    Text,
    Image,
} from 'react-native';
import PropTypes from 'prop-types';
import styles from './CommitStyle';

export default class Commit extends PureComponent {
    static propTypes = {
        data: PropTypes.objectOf(PropTypes.any).isRequired,
    };

    render() {
        const { data } = this.props;
        return (
            <View style={styles.container}>
                <Text style={styles.header}>{data.commit.message.split(/\r?\n/)[0]}</Text>
                <View style={styles.row}>
                    {data.author && data.author.avatar_url
                        ? <Image source={{uri: data.author.avatar_url}} style={styles.avatar} />
                        : null
                    }
                    <Text style={styles.sub}>
                        {data.author && data.author.login
                            ? <Text style={styles.bold}>{data.author.login}</Text>
                            : ''
                        } authored and <Text style={styles.bold}>{data.commit.committer.name.replace(' ', '_')}</Text></Text>
                </View>
            </View>
        );
    }
}

