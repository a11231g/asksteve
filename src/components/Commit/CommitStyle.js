import { StyleSheet } from 'react-native';

export default  StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: "#f9f9f9",
        padding: 15,
        borderBottomWidth: 1,
        borderColor: '#bbb',
        borderStyle: 'solid'
    },
    header: {
        fontWeight: 'bold',
        textAlign: 'left',
        fontSize: 12,
        marginBottom: 5
    },
    row: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    avatar: {
        width: 20,
        height: 20,
        marginRight: 5
    },
    sub: {
        fontSize: 10,
    },
    bold: {
        fontWeight: 'bold'
    }
})
