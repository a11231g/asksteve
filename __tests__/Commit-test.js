import { render, fireEvent } from 'react-native-testing-library';

import React from 'react';
import Commit from '../src/components/Commit/Commit';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const data = {
        author: {
            avatar_url: 'https://asd.com/asd.jpg',
            login: 'blah'
        },
        commit:{
            committer: {
                name: 'ali',
            },
            message: 'blah blah blah \n'
        }};
    const tree = renderer.create(<Commit data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
});
