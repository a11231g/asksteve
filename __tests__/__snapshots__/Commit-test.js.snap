// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f9f9f9",
      "borderBottomWidth": 1,
      "borderColor": "#bbb",
      "borderStyle": "solid",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 12,
        "fontWeight": "bold",
        "marginBottom": 5,
        "textAlign": "left",
      }
    }
  >
    blah blah blah 
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://asd.com/asd.jpg",
        }
      }
      style={
        Object {
          "height": 20,
          "marginRight": 5,
          "width": 20,
        }
      }
    />
    <Text
      style={
        Object {
          "fontSize": 10,
        }
      }
    >
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        blah
      </Text>
       authored and 
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        ali
      </Text>
    </Text>
  </View>
</View>
`;
